sdk/personalizer/ai-personalizer-rest/review/ai-personalizer.api.md

```diff
@@ -0,0 +1,936 @@
+## API Report File for "@azure-rest/ai-personalizer"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+/// <reference types="node" />
+
+import { Client } from '@azure-rest/core-client';
+import { ClientOptions } from '@azure-rest/core-client';
+import { HttpResponse } from '@azure-rest/core-client';
+import { KeyCredential } from '@azure/core-auth';
+import { RawHttpHeaders } from '@azure/core-rest-pipeline';
+import { RequestParameters } from '@azure-rest/core-client';
+import { StreamableMethod } from '@azure-rest/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public (undocumented)
+function createClient(Endpoint: string, credentials: TokenCredential | KeyCredential, options?: ClientOptions): PersonalizerClient;
+export default createClient;
+
+// @public (undocumented)
+export interface DateRangeOutput {
+    from: string;
+    to: string;
+}
+
+// @public (undocumented)
+export interface ErrorResponseOutput {
+    error: PersonalizerErrorOutput;
+}
+
+// @public (undocumented)
+export interface EvaluationContract {
+    enableOfflineExperimentation?: boolean;
+    endTime: Date | string;
+    name: string;
+    policies: Array<PolicyContract>;
+    startTime: Date | string;
+}
+
+// @public (undocumented)
+export interface EvaluationOutput {
+    creationTime: string;
+    endTime: string;
+    evaluationType: "Manual" | "Auto";
+    featureImportance?: Array<Array<string>>;
+    id: string;
+    jobId: string;
+    name: string;
+    optimalPolicy?: string;
+    policyResults?: Array<PolicyResultOutput>;
+    startTime: string;
+    status: "completed" | "pending" | "failed" | "notSubmitted" | "timeout" | "optimalPolicyApplied" | "onlinePolicyRetained";
+}
+
+// @public (undocumented)
+export interface EvaluationsCreate201Headers {
+    location?: string;
+}
+
+// @public
+export interface EvaluationsCreate201Response extends HttpResponse {
+    // (undocumented)
+    body: EvaluationOutput;
+    // (undocumented)
+    headers: RawHttpHeaders & EvaluationsCreate201Headers;
+    // (undocumented)
+    status: "201";
+}
+
+// @public (undocumented)
+export interface EvaluationsCreateBodyParam {
+    body: EvaluationContract;
+}
+
+// @public
+export interface EvaluationsCreatedefaultResponse extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseOutput;
+    // (undocumented)
+    status: string;
+}
+
+// @public (undocumented)
+export interface EvaluationsCreateMediaTypesParam {
+    contentType?: "application/json";
+}
+
+// @public (undocumented)
```

Comment:
Can you add some docs in the code, wherever it says undocumented?

Reply:
The swagger file does have a description for this as follows:
            "headers": {
              "Location": {
                "description": "Location of the Offline Evaluation status and data.",
                "type": "string"
              }
            }

This file is autogenerated. I do not author this file. @joheredi , is there something I should do to fix this?

Reply:
I think this is on the codegen side. I'll bring this up with the crew. IIRC we decided not to add comments for these parameters because they don't exist conceptually in Swagger, we generate them for organization, so there is not docs in the swagger we can use. Initially, we were writing something like

```ts
// Parameters for EvaluationsCreate operation
export type EvaluationsCreateParameters = EvaluationsCreateMediaTypesParam & EvaluationsCreateBodyParam & RequestParameters;
```

Which didn't really add any value, as it is pretty much the same as the name and we don't have a good way to generate a useful comment

Reply:
Thanks! Marking as resolved since this will be addressed external to this PR.